pyspeckit/cubes/SpectralCube.py

```diff
@@ -579,8 +579,12 @@ def get_modelcube(self, update=False):
         """
         if self._modelcube is None or update:
             yy,xx = np.indices(self.parcube.shape[1:])
+            isvalid = np.any(self.parcube, axis=0)
             self._modelcube = np.zeros_like(self.cube)
-            for x,y in zip(xx.flat,yy.flat):
+            # progressbar doesn't work with zip; I'm therefore giving up on
+            # "efficiency" in memory by making a list here.
+            for x,y in ProgressBar(list(zip(xx[isvalid],
+                                            yy[isvalid]))):
```

Comment:
This might be an improvement if you don't want to use lists:
```python
with ProgressBar(len(xx[isvalid])) as bar:
    for x,y in zip(xx[isvalid], yy[ixvalid]):
        self._modelcube[:,y,x] = self.specfit.get_full_model(pars=self.parcube[:,y,x])
        bar.update()
```